robot/impl/resource_manager.go

```diff
@@ -646,7 +666,7 @@ func (manager *resourceManager) completeConfig(
 		case <-ctx.Done():
 			return
 		}
-	}
+	} // for-each resource name
```

Comment:
Was intentional, if you find the comment acceptable. I find the control flow in `completeConfig` confusing due to the size of the for-loops (60 lines and a second 166 line). With the latter mostly* being a panic capturing Go lambda.

Reply:
ah that's fine as long as it was intentional. alternatively it would be good break the inner code down a bit if there any obvious refactors, but def beyond the scope of this PR